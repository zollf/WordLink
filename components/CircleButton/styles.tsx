import { StyleSheet } from "react-native";
import { widthPercentageToDP } from 'react-native-responsive-screen';

const styles = StyleSheet.create({
  button: {
    width: widthPercentageToDP('8%'),
    height: widthPercentageToDP('8%'),
    borderRadius: 17,
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    shadowColor: '#000',
    shadowRadius: 4,
    shadowOffset: { width: 2, height: 2},
    shadowOpacity: 0.1,
  },
  arrow: {
    width: widthPercentageToDP('4%'),
    height: widthPercentageToDP('4%'),
  },
});

export default styles;