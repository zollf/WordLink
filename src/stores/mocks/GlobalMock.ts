const GlobalMock: Global = {
  _ready: true,
  currentPage: 'index',
  setCurrentPage: jest.fn(),
  previousPage: 'index',
  currentDifficultyOpen: '',
  setCurrentDifficultyOpen: jest.fn((d: string) => (GlobalMock.currentDifficultyOpen = d)),
  getDifficultyColor: jest.fn(() => ''),
  setStartingPage: jest.fn(),
  onStartingPage: false,
  setUserInfo: jest.fn(),
  userInfo: {
    username: 'jeff',
    points: 20,
    puzzles: {
      completed: 2,
      easy: [],
      medium: [],
      hard: [],
    },
    hintsUsed: 6,
    hardestDifficulty: 'Medium',
    starsEarnt: 69,
  },
  init: jest.fn(),
};

export default GlobalMock;
