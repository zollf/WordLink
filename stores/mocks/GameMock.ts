const level: Level = {
  title: 'hello',
  difficulty: 'easy',
  game: {
    start: 'FORTY',
    end: 'FIFTY',
    path: 5,
  },
};

const GameMocks = {
  gameId: 0,
  difficulty: 'EASY',
  inGame: false,
  loaded: false,
  error: false,
  game: level.game,
  gameTitle: 'hello',

  currentWord: level.game.start,
  selectedLetter: '',
  stepNumber: 0,
  path: [level.game.start],
  visited: { FORTY: true },
  completed: false,

  loadGame: jest.fn(),
  startGame: jest.fn(),
  getNewLetters: jest.fn(() => ['A', 'B', 'C']),
  isWord: jest.fn(),
  setSelectedLetter: jest.fn(),
  changeLetter: jest.fn(),
  resetCurrentWord: jest.fn(),
  undo: jest.fn(),
  clearGame: jest.fn(),
};

export default GameMocks;
