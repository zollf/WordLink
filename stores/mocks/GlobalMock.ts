const GlobalMock: Global = {
  currentPage: 'index',
  setCurrentPage: jest.fn(),
  previousPage: 'index',
  currentDifficultyOpen: '',
  setCurrentDifficultyOpen: jest.fn((d: string) => (GlobalMock.currentDifficultyOpen = d)),
  getDifficultyColor: jest.fn(() => ''),
  setStartingPage: jest.fn(),
  onStartingPage: false,
};

export default GlobalMock;
