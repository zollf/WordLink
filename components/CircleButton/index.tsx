import React from 'react';
import { TouchableOpacity, Image, ImageSourcePropType } from 'react-native';
import { useStore } from '../../hooks';
import arrow from '../../images/left-arrow.png';
import close from '../../images/close.png';

import AppStyle from '../../styles';
import styles from './styles';

interface Props {
  color: string;
  overrideCallback?: () => void;
  symbol?: 'exit' | 'arrow'
}

const Button = ({ color, overrideCallback, symbol }: Props) => {
  const { setCurrentPage } = useStore('global');

  const handleClick = () => {
    if (overrideCallback) {
      overrideCallback();
    } else {
      setCurrentPage('menu');
    }
  };

  const image = () => {
    if (symbol) {
      if (symbol === 'exit') {
        return close;
      } else if (symbol === 'arrow') {
        return arrow;
      }
    } else {
      return arrow;
    }
  };

  return (
    <TouchableOpacity data-test-id="button" onPress={handleClick} style={[styles.button, {
      backgroundColor: AppStyle[color],
    }]}>
      <Image style={styles.arrow} source={image()} />
    </TouchableOpacity>
  );
};

export default Button;