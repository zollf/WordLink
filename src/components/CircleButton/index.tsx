import React, { useMemo } from 'react';
import { TouchableOpacity, Image } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';

import { useStore } from 'app/src/hooks';
import arrow from 'app/src/images/left-arrow.png';
import close from 'app/src/images/close.png';

import AppStyle from 'app/src/styles';
import styles from './styles';

interface Props {
  overrideCallback?: () => void;
  symbol?: 'exit' | 'arrow';
}

const Button = ({ overrideCallback, symbol }: Props) => {
  const { setCurrentPage } = useStore('global');

  const handleClick = () => {
    if (overrideCallback) {
      overrideCallback();
    } else {
      setCurrentPage('menu');
    }
  };

  const image = useMemo(() => {
    const images = {
      exit: close,
      arrow: arrow,
    };

    return symbol ? images[symbol] : arrow;
  }, []);

  return (
    <TouchableOpacity data-test-id="circle-button" onPress={handleClick}>
      <LinearGradient
        colors={[AppStyle.lightBlue, AppStyle.blue]}
        style={styles.button}
        locations={[0.5, 0.5]}
        start={[-0.1, 0.1]}
      >
        <Image style={styles.arrow} source={image} />
      </LinearGradient>
    </TouchableOpacity>
  );
};

export default Button;
