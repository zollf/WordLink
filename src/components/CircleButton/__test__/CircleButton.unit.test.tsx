import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'mobx-react';
import GlobalMock from 'app/src/stores/mocks/GlobalMock';
import GameMock from 'app/src/stores/mocks/GameMock';

import CircleButton from '..';

const onPress = jest.fn();

const component = (
  <Provider global={GlobalMock} gameStore={GameMock}>
    <CircleButton overrideCallback={onPress} symbol="exit" />
  </Provider>
);

describe('<CircleButton />', () => {
  let wrapper = renderer.create(component);
  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('click works correctly', () => {
    wrapper.root.findByProps({ 'data-test-id': 'circle-button' }).props.onPress();
    expect(onPress).toBeCalled();
  });

  it('default calls menu', () => {
    wrapper = renderer.create(
      <Provider global={GlobalMock} gameStore={GameMock}>
        <CircleButton />
      </Provider>,
    );
    wrapper.root.findByProps({ 'data-test-id': 'circle-button' }).props.onPress();
    expect(GlobalMock.setCurrentPage).toBeCalled();
  });
});
