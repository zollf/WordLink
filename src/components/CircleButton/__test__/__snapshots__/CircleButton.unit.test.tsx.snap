// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CircleButton /> matches snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4284860671,
          4282432255,
        ],
        "endPoint": undefined,
        "locations": Array [
          0.5,
          0.5,
        ],
        "startPoint": Array [
          -0.1,
          0.1,
        ],
      }
    }
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 30,
        "display": "flex",
        "height": 60,
        "justifyContent": "center",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 4,
        "width": 60,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
